schemes/hack_shallow/cmfmca.F90

```diff
@@ -0,0 +1,978 @@
+    real(kind_phys),    intent(in)  :: gravit       ! gravitational constant [m s-2]
+    real(kind_phys),    intent(in)  :: latvap       ! latent heat of vaporization [J kg-1]
+    real(kind_phys),    intent(in)  :: rhoh2o_in    ! density of liquid water [kg m-3]
+    real(kind_phys),    intent(in)  :: pref_edge(:) ! reference pressures at interface [Pa]
+
+    character(len=512), intent(out) :: errmsg
+    integer,            intent(out) :: errflg
+
+    ! local variables
+    integer :: k
+
+    errmsg = ''
+    errflg = 0
+
+    ! namelist variables
+    cmftau  = cmftau_in
+    c0      = c0_in
+
+    ! host model physical constants
+    cp      = cpair
+    rcp     = 1.0_kind_phys/cp
+    hlat    = latvap
+    rhlat   = 1.0_kind_phys/hlat
+    grav    = gravit
+    rgrav   = 1.0_kind_phys/gravit
+    rgas    = rair
+    rhoh2o  = rhoh2o_in
+
+    ! determine limit of shallow convection: regions below 40 mb
+    ! logic ported from convect_shallow_init with note that this calculation is repeated in the deep
+    ! convection interface.
+    if(pref_edge(1) >= 4.e3_kind_phys) then
+      limcnv = 1
+    else
+      limcnv = pver + 1
+      do k = 1, pver
+        if(pref_edge(k) < 4.e3_kind_phys .and. pref_edge(k+1) >= 4.e3_kind_phys) then
+          limcnv = k
+        endif
+      enddo
+    endif
+
+    ! in-module parameters (hardcoded)
+    dzmin = 0.0_kind_phys
+    betamn = 0.10_kind_phys
+
+    ! specify that relaxation timescale should be applied to column as opposed to triplets individually
+    rlxclm = .true.
+  end subroutine cmfmca_init
+
+!> \section arg_table_cmfmca_run Argument Table
+!! \htmlinclude cmfmca_run.html
+  subroutine cmfmca_run( &
+    ncol, pver, pcnst, &
+    const_props,       &
+    nstep, &
+    ztodt, &
+    pmid, pmiddry, &
+    pdel, pdeldry, rpdel, rpdeldry, &
+    zm, &
+    qpert_in, &
+    phis, &
+    pblh, &
+    t, &
+    q, & ! ... below are output arguments:
+    dq, &
+    qc_sh, &
+    cmfdt, &
+    cmfmc_sh, &
+    cmfdqr, &
+    cmfsl, &
+    cmflq, &
+    precc, &
+    cnt_sh, &
+    cnb_sh, &
+    icwmr, &
+    rliq_sh, &
+    scheme_name, &
+    flx_cnd, &
+    errmsg, errflg &
+  )
+    ! framework dependency for const_props
+    use ccpp_constituent_prop_mod, only: ccpp_constituent_prop_ptr_t
+
+    ! to_be_ccppized
+    use wv_saturation,             only: qsat
+
+    ! temporary for debug
+    use cam_logfile,               only: iulog
+
+    ! Input arguments
+    integer,         intent(in)     :: ncol               ! number of atmospheric columns
+    integer,         intent(in)     :: pver               ! number of vertical levels
+    integer,         intent(in)     :: pcnst              ! number of ccpp constituents
+    type(ccpp_constituent_prop_ptr_t), &
+                     intent(in)     :: const_props(:)     ! ccpp constituent properties pointer
+    integer,         intent(in)     :: nstep              ! current time step index
```

Comment:
I don't think this is used - can remove from argument list and metadata.

Reply:
Done, thanks!